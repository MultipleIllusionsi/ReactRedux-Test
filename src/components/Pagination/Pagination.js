import React from "react";
import PropTypes from "prop-types";

import "./Pagination.scss";

const Pagination = ({ length, page, handler }) => {
  const numberOfPage = Math.ceil(length);
  const arr = [];

  for (let i = 1; i <= numberOfPage; i++) {
    arr[i - 1] = i;
  }

  return (
    <ul onClick={handler} className="pagination">
      {arr.map((elem, idx) => (
        <li
          key={idx}
          value={elem}
          className={`pagination__item ${page === elem && "selected"}`}
        >
          {elem}
        </li>
      ))}
    </ul>
  );
};

Pagination.propTypes = {
  length: PropTypes.number.isRequired,
  page: PropTypes.number.isRequired,
  handler: PropTypes.func.isRequired
};

export default Pagination;
