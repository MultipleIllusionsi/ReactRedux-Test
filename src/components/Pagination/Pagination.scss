.pagination {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    margin-top: 7vh;

    &__item {
        padding: 1rem 2rem;
        border: 1px solid #a4b0be;
        font-size: 2rem;
        transition: 0.2s all ease-out;
        cursor: pointer;

        &:hover {
            background-color: lighten($color: #bdc3c7, $amount: 10);
            transform: scale(1.05);
        }
    }

    & .selected {
        background-color: #bdc3c7;
        transform: scale(1.05);
    }
}