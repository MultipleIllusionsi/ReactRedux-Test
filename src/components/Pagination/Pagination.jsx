import React from "react";

import "./Pagination.scss";

const Pagination = ({ length, page, handler }) => {
  const numberOfPage = Math.ceil(length);
  const arr = [];
  for (let i = 1; i <= numberOfPage; i++) {
    arr[i - 1] = i;
  }

  return (
    <ul onClick={handler} className="pagination">
      {arr.map((elem, idx) => (
        <li
          key={idx}
          value={elem}
          className={`pagination__item ${page === elem && "selected"}`}
        >
          {elem}
        </li>
      ))}
    </ul>
  );
};

export default Pagination;
