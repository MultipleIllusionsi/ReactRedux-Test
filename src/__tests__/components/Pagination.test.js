import { shallow } from "enzyme";
import React from "react";
import Pagination from "../../components/Pagination/Pagination";

it("expect to render Pagination component", () => {
  const component = shallow(
    <Pagination handler={() => {}} page={5} length={10} />
  );
  expect(component.debug()).toMatchSnapshot();
});
