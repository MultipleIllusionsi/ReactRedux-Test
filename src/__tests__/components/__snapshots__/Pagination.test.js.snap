// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Pagination component 1`] = `
"<ul onClick={[Function: handler]} className=\\"pagination\\">
  <li value={1} className=\\"pagination__item false\\">
    1
  </li>
  <li value={2} className=\\"pagination__item false\\">
    2
  </li>
  <li value={3} className=\\"pagination__item false\\">
    3
  </li>
  <li value={4} className=\\"pagination__item false\\">
    4
  </li>
  <li value={5} className=\\"pagination__item selected\\">
    5
  </li>
  <li value={6} className=\\"pagination__item false\\">
    6
  </li>
  <li value={7} className=\\"pagination__item false\\">
    7
  </li>
  <li value={8} className=\\"pagination__item false\\">
    8
  </li>
  <li value={9} className=\\"pagination__item false\\">
    9
  </li>
  <li value={10} className=\\"pagination__item false\\">
    10
  </li>
</ul>"
`;
